### ISSUE TITLE
Search with -products commandline option works only by full ID, contrary to description

### ISSUE THREAD

**Issue by abelbraaksma** [Microsoft/vswhere#114]
The description in the help says:

    -products arg  One or more products to find. Defaults to Community, Professional, and Enterprise.
                   Specify "*" by itself to search all product instances installed.
                   See https://aka.ms/vs/workloads for a list of products.

This suggests you can use `-products Enterprise` to find enterprise products, which is not how it works. I noticed it will only find a product by its full ID, i.e. the following command will find Enterprise products:

    vswhere -products Microsoft.VisualStudio.Product.Enterprise

I suggest to either update the helptext, for instance by specifying it requires the full ID and/or give an example string, or change the way it behaves by searching for substrings, or end-of-strings. The latter being enough specific to work with a string like `Enterprise`.

**Comment by heaths** [comment]
Given the link at the bottom of the text has the full IDs, you should use that. The shortened text are akin to the editions (which was what was intended), and increasing the binary by almost 200 bytes is not worth it. This tool is geared more toward detection scripts and not meant to be called often where allowing for a shortened string might be useful.

**Comment by abelbraaksma** [comment]
@heaths, my issue is mainly with the description and if the number of bytes are a dealbreaker, perhaps we can settle on clarifying the text as follows?

```
-products arg  Finds instances matching the give product IDs, defaults to Community, 
               Professional and Enterprise. "*" searches for all installed product IDs.
               See https://aka.ms/vs/workloads for a list of products.
```

I think this is clearer, and does not add bytes to the binary.

**Comment by heaths** [comment]
1. You forgot the Oxford comma. I would never accept such a PR without one. ;)
2. I don't actually see how that's any better. You still only mention the editions. If your main goal is to just add "ID" mainly for clarification I could add that, but I wouldn't ship a new version just for that. It would get picked up whenever we release another version.

**Comment by abelbraaksma** [comment]
> You forgot the Oxford comma. I would never accept such a PR without one. ;)

Oh oh, I actually edited it to remove it. In Dutch it's frowned upon. Learned something new today, thanks!

> You still only mention the editions. If your main goal is to just add "ID" mainly for clarification 

Yes, exactly (otherwise it gets unwieldy, I understand that). It's mainly programmers or IT staff that use tools like these. They are triggered by exactness in phrases, or they will use trial and error to find out how it works. So yes, "just adding ID" could help a lot, I think. 

Disclaimer: I participated for years in creating the XSLT 3.0 spec. We often had lengthy debates about such seemingly simple things as word-smithing for concise, readable definitions. It's not that a big deal here, obviously, but I think it is still an improvement. Definitely not worthy of a separate release, of course, I didn't intend to suggest that ;).
